Add base aa1 to master

kleberbaum opened the pull request:
**Checklist:**

 <!--- Make sure your PR is documented and tested before submission. Put an `x` in all the boxes that apply: -->
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?


**Type of Changes:**

 <!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)


**What is the current behavior? (link to any open issues here)**
- Currently package.json does not satisfy our needs. #1 
- Currently code of conduct does not satisfy our needs. #2 
- Currently `.gitignore` does not satisfy our needs. #3 
- Currently components are hindering our progress. #4  
- Currently the README is still a bit generic. #6 
- Currently there is no COPYING file. #7 
- Currently the licenses are not up to date. #8
- Currently there are no import and export anchors. #9
- Currently there is no basic components folder. #10
- Currently the Visual Studio code settings are outdated. #11

**What is the new behavior (if this is a feature change)?**
- Now package.json has been updated. Closes #1 
- Now code of conduct has been updated. Closes #2 
- Now `.gitignore` files has been updated. Closes #3 
- Now components have been removed. Closes #4 
- Now the README is no longer generic. Closes #6 
- Now there is a COPYING file. Closes #7 
- Now the licenses are up to date. Closes #8 
- Now there are import and export anchors. Closes #9 
- Now there is a basic components folder. Closes #10
- Now the Visual Studio Code settings are up to date. Closes #11


**Other information**:
- The version has been updated to v0.1.0-alpha to start this project.
- Ref: https://github.com/snek-at/front#versioning

Aichnerc committed:
Update code of conduct

The code of conduct has been updated to our new
standards.

Aichnerc committed:
Update package

The package has been updated to fit this project.

Aichnerc committed:
Update gitignore

The gitignore has been updated to fit this project.
The .env files and mdb react pro file have been
excluded.

Aichnerc committed:
Remove components

The components have been removed due to #4

Aichnerc committed:
Update scss color variables

The SCSS color variables have been updated to fit
the project.

Aichnerc committed:
Improve code quality

A space has been removed.

Aichnerc committed:
Fix package.json

A typo has been fixed that caused the npm install to fail.

Aichnerc committed:
Fix SCSS

A non-existent scss file has been removed.

Aichnerc committed:
Update Routes and App

Non existent components have been excluded for now.

kleberbaum commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,61 @@
+- No commits, under no circumstances, should be directly pushed to master (deadly)
+- Each commit should only serve one purpose
+
+## [](#pull-requests)Pull Requests
```
Wrong file please move to CONTRIBUTING.md.

kleberbaum commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,61 @@
 be high on both sides of the process.  Try to keep in mind the immortal
 words of Bill and Ted, "Be excellent to each other."
+
+## [](#commits)Commits
```
Wrong file please move to CONTRIBUTING.md.

kleberbaum commented on package.json:
```diff
@@ -1,10 +1,11 @@
-  "name": "react-mdb-template",
-  "version": "1.0.2",
+  "name": "snek-front",
+  "version": "v1.0.0-alpha",
```
xD nope
The package.json should only contain the version number not the version tag.

kleberbaum commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,61 @@
-Code of Conflict
-----------------
```
Would like to keep the headline.

kleberbaum commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,61 @@
-criticism.  The review will almost always require improvements to the
-code before it can be included.  Know that this happens
-because everyone involved wants to see the best possible solution for
-the overall success of this project.
```
Wouldn't change this at all.

kleberbaum commented on src/Routes.js:
```diff
@@ -8,15 +8,17 @@ import { Route, Switch } from 'react-router-dom';
-} from './components/pages';
+//import {
+//  HomePage
+//} from './components/pages';
```
Wouldn't it be better to comment out multiple lines w/
/*
*/

schettn commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,61 @@
-criticism.  The review will almost always require improvements to the
-code before it can be included.  Know that this happens
-because everyone involved wants to see the best possible solution for
-the overall success of this project.
```
Please undo your whitespace changes to fit the format of the paragraph below.

Aichnerc replied:
Yeah, I agree. There should only be one space after each sentence, even if two spaces make the text more readable.
However, we would need to include it to our guidelines, which would lead to exzessive usage - which leads to habits.
And this one isn't a good one for me.

kleberbaum commented:
Please consider to link some references of our versioning convention at "other information".

Aichnerc committed:
Update CONTRIBUTING.md

The contributing file has been updated due to
a request by @kleberbaum

Aichnerc committed:
Update CODE_OF_CONDUCT.md

The code of conduct has been changed due to
a request by @kleberbaum

Aichnerc committed:
Update version

The version has been updated due to a request by
@kleberbaum.

Aichnerc committed:
Improve code quality

The commenting style of Routes components has
been changed due to a request of @kleberbaum

Aichnerc committed:
Update CODE_OF_CONDUCT.md

The whitespaces in the last paragraph have been removed
due to a request of @schettn

kleberbaum committed:
Improve code quality

The code quality has been
improved due to the codacy check.

kleberbaum reviewed (commented):
Should be ok to merge.

petschna reviewed (approved):
Should be ok to merge.

schettn committed:
Improve .vscode/settings.json

The VSCode editor rulers are added now in order to prevent
line overflow 80c, 120c. Also formatting applies now on saving.

schettn committed:
Add basic components

The basic structure for the components
was added according to atom structure.

schettn committed:
Add COPYING

A COPYING file has been added.
Ref: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/COPYING

schettn committed:
Change License

A new licence tag has been added to all files.

schettn committed:
Add import/export anchors

Import and export anchors have been
added to necessary files.

schettn committed:
Update PULL_REQUEST_TEMPLATE

The PULL_REQUEST_TEMPLATE was updated
due to an new "type of change" option.

schettn committed:
Update README.md

The template readme has been updated to
fit needs of front. Also the creators have been
adjusted.

pinterid commented on .vscode/settings.json:
```diff
@@ -1,42 +1,53 @@
-        "renamer",
-        "webfonts"
-    ]
+  "editor.fontFamily": "Fira Code, Consolas, 'Courier New', monospace",
```
Add these settings:
 - "prettier.trailingComma": "es5",
 - "prettier.arrowParens": "always",

pinterid replied:
Just copy the .prettierrc file from intel. This will update the settings for prettier.

schettn replied:
I would suggest to implement this like intel does:
https://github.com/snek-at/intel/pull/77#discussion_r436252307

pinterid commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,46 @@
+- Each pull request must have at least one assigned developers
+- Each pull request must have a label
+- Each pull request should be merged by a reviewer
+
```
Remove newline.

pinterid commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,46 @@
+- [Table of contents](#table-of-contents)
+- [[](#commits)Commits](#commitscommits)
+- [[](#pull-requests)Pull Requests](#pull-requestspull-requests)
+
```
Remove newline.

schettn committed:
Improve Import Regions

Internal and external imports are separated
now by a new line.

pinterid commented on README.md:
```diff
@@ -99,7 +138,6 @@ to [contribute to our project](#contributing).
-Copyright © 2019 Werbeagentur Christian Aichner
+Copyright © Simon Prast
 
 [discord-badge]: https://img.shields.io/badge/Discord-Join%20chat%20%E2%86%92-738bd7.svg
```
The badge should also be removed.

schettn commented on package.json:
```diff
@@ -1,10 +1,11 @@
-  "name": "react-mdb-template",
-  "version": "1.0.2",
+  "name": "snek-front",
+  "version": "1.0.0",
```
Wrong version number. Change it to:
```suggestion
  "version": "0.1.0",
```

schettn commented on package.json:
```diff
@@ -1,10 +1,11 @@
   "license": "(EUPL-1.2)",
   "dependencies": {
-    "mdbreact": "^4.19.1",
+    "mdbreact": "mdbreact-4.20.0.tgz",
```
This version is outdated. Change it to:
```suggestion
    "mdbreact": "mdbreact-4.26.1.tgz",
```

pinterid reviewed (changes requested):
Looks good after some changes.

Aichnerc commented on .gitignore:
```diff
@@ -7,9 +7,22 @@ package-lock.json
+.env.production.local
+
+# MDB
+mdbreact-*
```
Please change to `mdb*`. Some MDB ressources have names like `mdb-sortable-*`
```suggestion
mdb*
```

Aichnerc commented on README.md:
```diff
@@ -1,34 +1,30 @@
-  <a href="https://www.aichner-christian.com/" target="_blank" rel="noopener noreferrer">
-    <img src="https://www.aichner-christian.com/img/logo/logo_web.png" alt="Agency Logo" height="150">
+  <a href="https://snek.at/" target="_blank" rel="noopener noreferrer">
+    <img src="https://avatars2.githubusercontent.com/u/55870326?s=400&u=c6c7f06305ddc94747d474850fde7b2044f53838&v=4" alt="Agency Logo" height="150">
```
This is not the Agency logo.
```suggestion
    <img src="https://avatars2.githubusercontent.com/u/55870326?s=400&u=c6c7f06305ddc94747d474850fde7b2044f53838&v=4" alt="SNEK Logo" height="150">
```

pinterid replied:
Change "Agency" to "SNEK".

schettn commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,46 @@
+  <a href="https://www.aichner-christian.com/" target="_blank" rel="noopener noreferrer">
+    <img src="https://www.aichner-christian.com/img/logo/logo_web.png" alt="Agency Logo" height="150">
+  </a>
+</p>
```
There shouldn't be Aichner stuff.

schettn reviewed (changes requested):
Please check

schettn committed:
Update package.json

Change version tag

schettn committed:
Update package.json

The mdb version has been updated to "4.26.1"

Aichnerc commented on src/Routes.js:
```diff
@@ -1,23 +1,30 @@
 //> Components
 /**
  * HomePage: A basic template page
  */
```
This description is no longer needed, once it serves little purpose. It should be reworked with `@description` for each and every imported page.

Aichnerc commented on src/Routes.js:
```diff
@@ -1,23 +1,30 @@
 import {
   HomePage
 } from './components/pages';
+*/
```
I understand that this is the base, but to keep it clean please remove all unused imports.

Aichnerc commented on src/Routes.js:
```diff
@@ -1,23 +1,30 @@
+        {
+          //<Route exact path='/' component={HomePage} />
+        }
+
```
Remove unused code. This is not the proper way to declare Routes anyways.

schettn committed:
Update CONTRIBUTING.md

Wrong links have been updated to SNEK

pinterid committed:
Add Prettier config file

The configuration file for prettier was added.

Aichnerc commented on CONTRIBUTING.md:
```diff
@@ -1,9 +1,46 @@
-# Contributing
+<p align="center">
+  <a href="https://snek.at/" target="_blank" rel="noopener noreferrer">
+    <img src="https://avatars2.githubusercontent.com/u/55870326?s=400&u=c6c7f06305ddc94747d474850fde7b2044f53838&v=4" alt="Agency Logo" height="150">
```
```suggestion
    <img src="https://avatars2.githubusercontent.com/u/55870326?s=400&u=c6c7f06305ddc94747d474850fde7b2044f53838&v=4" alt="SNEK Logo" height="150">
```

pinterid replied:
Change "Agency" to "SNEK".

Aichnerc commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,26 @@
-----------------
+<p align="center">
+  <a href="https://www.aichner-christian.com/" target="_blank" rel="noopener noreferrer">
+    <img src="https://www.aichner-christian.com/img/logo/logo_web.png" alt="Agency Logo" height="150">
```
```suggestion
    <img src="https://www.aichner-christian.com/img/logo/logo_web.png" alt="SNEK Logo" height="150">
```

Aichnerc commented on package.json:
```diff
@@ -13,11 +14,10 @@
   "repository": {
     "type": "git",
-    "url": "https://github.com/aichner/React-MDB-Template.git"
+    "url": "https://github.com/snek-at/front.git"
```
Can't you use this fancy method here too?

```suggestion
    "url": "git+https://github.com/snek-at/front.git"
```

Aichnerc commented on package.json:
```diff
@@ -13,11 +14,10 @@
     "start": "react-scripts start",
-    "build": "react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject"
```
Add required scripts.

```suggestion
    "build": "react-scripts build",
    "predeploy": "yarn run build",
    "deploy": "gh-pages -d build",
    "eject": "react-scripts eject"
```

pinterid commented on CODE_OF_CONDUCT.md:
```diff
@@ -1,15 +1,26 @@
-Code of Conflict
-----------------
+<p align="center">
+  <a href="https://www.aichner-christian.com/" target="_blank" rel="noopener noreferrer">
```
Remove aichner stuff from this file

Aichnerc commented on src/components/atoms/index.js:
```diff
@@ -1,16 +1,14 @@
-//> Components
```
There are no atoms yet. Please remove the folder `src/components/atoms` and re-add it as soon as you create atoms.

pinterid replied:
We preferred to create the structure with this branch, to make further contributing easier.

Aichnerc commented on src/components/molecules/index.js:
```diff
@@ -1,14 +1,14 @@
-//> Components
```
There are no molecules yet. Please remove the folder `src/components/molecules` and re-add it as soon as you create molecules.

pinterid replied:
We preferred to create the structure with this branch, to make further contributing easier.

Aichnerc commented on src/components/pages/index.js:
```diff
@@ -1,12 +1,14 @@
-//> Components
```
There are no pages yet. Please remove the folder `src/components/pages` and re-add it as soon as you create pages.

pinterid replied:
We preferred to create the structure with this branch, to make further contributing easier.

Aichnerc reviewed (changes requested):
There are a few things I've noticed. 🧶

pinterid committed:
Improve documentation quality

The documentation quality was improved due to a request by @pinterid.

pinterid committed:
Update package.json

The package.json scripts were updated, so the basic commands can be used again.

pinterid committed:
Update .gitignore

The .gitignore file was updated so it ignores all kind of mdb* stuff.

schettn committed:
Improve code quality

The quality of the code has been improved
due to a request of @aichnerc

pinterid committed:
Merge remote-tracking branch 'origin/add-base-aa1' into add-base-aa1

pinterid committed:
Add folder structure organisms

The folder structure for organisms was added.

pinterid committed:
Update package.json

The package.json was updated due to a request by @aichnerc.

schettn commented on .prettierrc:
```diff
@@ -0,0 +1,6 @@
+  "arrowParens": "always",
+  "tabWidth": 2,
+  "singleQuote": false,
+}
```
Please add a newline:
```suggestion
}

```

schettn committed:
Update .prettierrc

Add a newline to the file end.

Aichnerc commented on package.json:
```diff
@@ -13,12 +14,13 @@
   },
   "repository": {
     "type": "git",
```
Is the type still required if you include `git+` in the url?

schettn reviewed (approved):
LGTM 🚀

Aichnerc commented on src/App.js:
```diff
@@ -1,41 +1,48 @@
+import { BrowserRouter as Router } from "react-router-dom";
 
 //> Components
 /**
  * Footer: Global Footer
  * Navbar: Global navigation bar
  */
```
Consider using `@description`

pinterid reviewed (approved):
Looks good.

Aichnerc commented on src/Routes.js:
```diff
@@ -1,23 +1,19 @@
-import {
-  HomePage
-} from './components/pages';
+// import { SamplePage } from "./components/pages";
```
Consider creating a sample `@description` to make it easier to not forget to add documentation later.

Aichnerc reviewed (approved):
Feel free to look at my suggestions, but the code looks ok nevertheless.

pinterid committed:
Update CODE_OF_CONDUCT.md

The CODE_OF_CONDUCT.md was changed due to a request by Codacy.

pinterid committed:
Update CONTRIBUTING.md

Update CONTRIBUTING.md due to a request by Codacy.

pinterid committed:
Update README.md

The README.md was changed due to a request by Codacy.

pinterid committed:
Update README.md

The README.md was changed due to a request by Codacy.
